### PR TITLE
DAOS-6180 control: Instrument agent shutdown

### PR DESCRIPTION
**Pull request by mjmac**
Adds some timing information to measure the time between
shutdown signal received and shutdown completion. This
should help to diagnose intermittent CI failures where
daos_agent did not stop within the expected amount of
time. Updates the stop_processes() harness method to
send an ABRT signal before sending KILL in order to
dump stack traces for stuck go processes.

Also makes the server/agent startup messages more consistent
in style and information by using the formal component names
and including the pid in addition to version.

### CODE REVIEW

**Review by tanabarr — APPROVED** [review]

**Comment by mjmac** [comment]
The result of this change is that the agent logs look like this now (note that most lines are DEBUG, so only the most pertinent details will be printed in production):
```
DEBUG 06:03:08.766076 start.go:50: Starting DAOS Agent v1.1.2 (pid 14428)
DEBUG 06:03:08.766151 start.go:57: Full socket path is now: /var/run/daos_agent/daos_agent.sock
DEBUG 06:03:08.786181 start.go:102: startup complete in 20.001488ms
agent INFO 2020/12/09 06:03:08 DAOS Agent v1.1.2 (pid 14428) listening on /var/run/daos_agent/daos_agent.sock
agent INFO 2020/12/09 06:03:37 Signal received.  Caught terminated; shutting down
DEBUG 06:03:37.397137 start.go:131: shutdown complete in 147.523Âµs
```

**Review by phender — CHANGES_REQUESTED** [review]
If I'm understanding the intent of the changes correctly, then I believe a change is needed.  If not, then the python code changes look good.

File: src/tests/ftest/util/general_utils.py [hunk 1]
```diff
@@ -495,9 +495,13 @@ def stop_processes(hosts, pattern, verbose=True, timeout=60):
             "sudo pkill {}".format(pattern),
             "sleep 5",
             "if pgrep --list-full {}".format(pattern),
+            "then pkill --signal ABRT {}".format(pattern),
+            "sleep 1",
+            "if pgrep --list-full {}".format(pattern),
```
Comment (phender) [inline src/tests/ftest/util/general_utils.py:500]: The normal agent and server stopping process goes through [`stop()`](https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/command_utils.py#L187).  Here a INT signal is sent and, if needed, it is followed by a KILL signal, as controlled by [this list](https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/command_utils.py#L197).

The `stop_processes()` method is only called when
1. ensuring there are no existing server processes running on the hosts on which the server is about to be started ([here](https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/server_utils.py#L361))
2. an error is detected during server startup ([here](https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/server_utils.py#L487) and [here](https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/server_utils.py#L507))
3. as a catch-all for any processes not stopped by `stop()` ([here](https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/server_utils.py#L596) and [here](https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/agent_utils.py#L205))

In typically operation, even though `stop_processes()` is called it is essentially a no-op as the initial `prep` in the command string does not match anything.

I suspect that the `ABRT` signal should instead be added to [`stop()`](https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/command_utils.py#L197) to get the intended result.

**Review by phender — COMMENTED** [review]
In my previous review I neglected to add my comment.  Here they are.

File: src/tests/ftest/util/general_utils.py (hunk 1 above)
Comment (mjmac) [reply src/tests/ftest/util/general_utils.py:500]: Hmm. But `stop()` doesn't send SIGTERM to (daos_agent|daos_server), does it? It sends it to the parent orterun process. I don't think that sending SIGABRT to orterun would be passed down its child process(es).

The intent behind adding the SIGABRT signal here was to provide additional information in diagnosing whatever happened in https://jira.hpdd.intel.com/browse/DAOS-6180, which if I understand correctly has already gotten to `stop_processes()`.

**Review by kjacque — APPROVED** [review]
The Go changes LGTM.

**Pull request merged**